test/shell-for/default.nix

```diff
@@ -0,0 +1,45 @@
+{ stdenv, util, mkStackSnapshotPkgSet }:
+
+with stdenv.lib;
+with util;
+
+let
+  pkgSet = mkStackSnapshotPkgSet {
+    resolver = "lts-12.21";
+    pkg-def-extras = [(hackage: {
```

Comment:
I think we hsould add a comment why these custom packages are necessary.  Eventually someone will be hopelessly lost as to what the hell is going on.

Best to link the relevant stackage issue: commercialhaskell/stackage#4466

Reply:
Good idea, commented. That issue suggests that we could fix the yaml in stackage.
One thing I'm confused about is that this is a GHC 8.4.4 LTS not 8.6.4.

Reply:
That is indeed a curious case. Does it work without the extras argument? This change seemed to be necessary for 8.6.x only.

Reply:
Without the extras argument, it fails with:
```
Configuring library for transformers-0.5.5.0..
Setup: Encountered missing dependencies:
base (>=2 && <6) && <0
```
For the ghc-8.4.4 which is used by the build, `ghc-pkg list | grep transformers` reports 0.5.5.0, and it's the same version in [lts-12.21](https://www.stackage.org/lts-12.21/package/transformers-0.5.5.0). It looks like there is a [hackage cabal revision](http://hackage.haskell.org/package/transformers-0.5.5.0/transformers.cabal) that disqualifies this package in the cabal solver.

Reply:
what a mess...

Since https://gitlab.haskell.org/ghc/ghc/issues/16199#note_174847 (and the script was added to CI) I hope that going forward 8.6+, ghc doesn't lie about the shipped packages anymore.